example/bwmf/proto/bwmf.proto

```diff
@@ -15,19 +15,14 @@ message Request {
 
 message Response {
   uint64 blockId = 1;
-  DenseMatrixShard shard = 2;
+  MatrixShard shard = 2;
 }
 
-message DenseMatrixShard {
-  message DenseRow {
-    repeated float at = 1;
+message MatrixShard {
+  message Row {
+    uint32 rowId = 1;
+    repeated uint32 colIds = 2;
+    repeated float values = 3;
   }
-  repeated DenseRow row = 1;
-}
-
-message SparseMatrixShard {
-  message SparseRow {
-    map<int32, float> at = 1;
-  }
-  repeated SparseRow row = 1;
+  repeated Row rows = 1;
 }
```

Comment:
I was think something like
````
message MatrixShard {
  repeated bool sparse = 1;
  repeated float data = 2;
  repeated uint32 ia = 3;
  repeated uint32 ja = 4;
}
````

for IA, JA see: http://en.wikipedia.org/wiki/Sparse_matrix#Yale

The advantage of this is that we can use linear algebra library in the future (at least for dense matrix), by just passing the data field to the library.